src/ray/core_worker/core_worker.cc

```diff
@@ -516,6 +520,13 @@ CoreWorker::CoreWorker(const CoreWorkerOptions &options, const WorkerID &worker_
         }
       },
       100);
+
+#ifndef _WIN32
+  // nice() increments the niceness of this process, so this assumes one CoreWorker object
+  // per process.
+  const auto niceness = nice(RayConfig::instance().worker_niceness());
```

Comment:
Please add a note that we should only do this after the worker has registered to the raylet (otherwise we could thrash trying to start workers when under pressure).